nixos/modules/services/audio/spotifyd.nix

```diff
@@ -0,0 +1,47 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.spotifyd;
+  spotifydConf = pkgs.writeText "spotifyd.conf" cfg.config;
+in
+{
+  options = {
+    services.spotifyd = {
+      enable = mkEnableOption "spotifyd, a Spotify playing daemon";
+
+      config = mkOption {
```

Comment:
Would you be up for replacing this with an attribute set? If so I would really love to see a `username` and `passwordFile` option added to this so the user would normally never need to specify any `config` values at all.

Reply:
Sure, if there is a use case for supplying username and password to spotifyd? I use spotifyd as a Spotify Connect client (server?) which I can control from any Spotify client on my network (no matter who is logged in on that Spotify client). For this I do not need to supply username and password to spotifyd. Per the official repo https://github.com/Spotifyd/spotifyd#d-bus-mpris one can control spotifyd via "D-Bus MPRIS" but this feature is experimental and needs to be enabled before compilation. Are there other use cases I am ignorant of? Or perhaps things are not always as easy in my use case as they are for me?

Reply:
I didn't realize this could be run without credentials. Sorry. My comment was based on that incorrect assumption.